pkg/istio-agent/xds-proxy.go

```diff
@@ -141,17 +153,42 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 			// todo close downstream?
 			return err
 		case req := <-requestsChan:
+			if req.TypeUrl == v3.ListenerType {
+				// for every LDS req, add a NDS req
```

Comment:
Why do we piggyback on LDS, rather than just originating our own NDS request at the start of each stream?

Reply:
its not a one time thing right? NDS gets updated everytime a new service is added/removed.. I think we can do it - by creating our own nonce, and creating a req after each response. this felt _convenient_ :).

Reply:
A request is one time though. Istiod should push and updates, we just send 1 initial request to subscribe?

Reply:
I see now you are using this to ACK in addition to the original request. This feels more wrong, as we are now ACKing NDS when we mean to ACK LDS. Seems bit off